plugins/text-freetype2/text-functionality.c

```diff
@@ -464,9 +464,8 @@ void load_text_from_file(struct ft2_source *srcdata, const char *filename)
 		bfree(srcdata->text);
 		srcdata->text = NULL;
 	}
-	srcdata->text = bzalloc((strlen(tmp_read) + 1) * sizeof(wchar_t));
-	os_utf8_to_wcs(tmp_read, strlen(tmp_read), srcdata->text,
-		       (strlen(tmp_read) + 1));
+	srcdata->text = bzalloc((filesize + 1) * sizeof(wchar_t));
```

Comment:
Please measure the size of the decoded wcs data by calling `os_utf8_to_wcs` without a destination pointer.

Reply:
And use that as the text size instead of filesize.

Reply:
```suggestion
	textsize = os_utf8_wcs(tmp_read, filesize, NULL, 0);
	srcdata->text = bzalloc((textsize+1) * sizeof(wchar_t));
	os_utf8_to_wcs(tmp_read,filesize,srcdata->text,(textsize+1));
```

All good?

Reply:
Yes something like that

Reply:
Should I create the textsize variable at the top along with others?